lib/kernel/doc/guides/socket_usage.md

```diff
@@ -524,3 +524,69 @@ Options for level `sctp`:
 | rtoinfo           | sctp_rtoinfo()         | yes | yes | none                            |
 
 _Table: sctp options_
+
+
+[](){: #socket_configure_flags }
+
+## Socket Configure Flags
+
+There are a couple of configure flags, that can be used when (configure and)
+building Erlang/OTP, which effect the functionality of the 'socket' nif.
+
+### Builtin Socket Support
+
+Support for the builtin 'socket' (as a nif) can be explicitly enabled and
+disabled:
+
+```text
+--enable-esock (default) | --disable-esock
+```
+
+### RCVTIMEO/SNDTIMEO socket options
+
+Support for these (socket) options has to be explicitly enabled (see why above):
+
+```text
+--enable-esock-rcvsndtimeo | --disable-esock-rcvsndtimeo (default)
+```
+
+### Extended Error Info
+
+The use of Extended Error Info (currently only used on Windows) can be
+explicitly enabled and disabled:
+
+```text
+--enable-esock-extended-error-info (default) | --disable-esock-extended-error-info
+```
+
+### Verbose Mutex Names
+
+The use of verbose mutex names (in the 'socket' nif) can be explicitly
+enabled and disabled:
+
+```text
+--enable-esock-verbose-mtx-names (default) | --disable-esock-verbose-mtx-names
+```
```

Comment:
It is unclear to me why anyone should care about this, is it a memory/debug tradeoff?

Reply:
The default name for a mutex is something like: "esock.r[10]" (the '10' is the file descriptor).
This is good for debug, but in some uses cases this can be costly.
So the name can be simplified to something like: "esock.r". Its not unique, but its "cheeper".
But as your question indicates, this should be part of the description.

Reply:
Maybe "Therefore a simplified name scheme without FD  information can be used to avoid the bottleneck but loosing some information that can be useful for debugging"  Current formulation I think concentrates to much on the implementation and too little on how it affects the user.

Reply:
Its *not* a bottleneck in normal use. The bottleneck *only* occurs for very specific debug scenarios.

Reply:
Ok, so it is a debug tradeoff only. I guess we want to make that clear then.